src/rules.ts

```diff
@@ -43,7 +123,13 @@ function evaluateRuleConditions(
 ): boolean[] {
   return conditions.map((condition) =>
     obfuscated
-      ? evaluateObfuscatedCondition(subjectAttributes, condition)
+      ? evaluateObfuscatedCondition(
+          Object.entries(subjectAttributes).reduce(
+            (accum, [key, val]) => ({ [getMD5Hash(key)]: val, ...accum }),
```

Comment:
Random thought: We don't salt when we hash. This could subject us to rainbow attacks. Although our salt would also be public, so probably not helpful to do so.

Reply:
I went through the exact same thought process...